src/org/exist/dom/memtree/InMemoryXMLStreamReader.java

```diff
@@ -74,6 +80,11 @@ public int next() throws XMLStreamException {
 
                 if(next < 0) { // no child nodes
                     state = XMLStreamReader.END_ELEMENT;
+
+                    if (state == XMLStreamReader.END_ELEMENT) {
```

Comment:
@adamretter the same value has been added two lines up.....  so maybe a } is misplaced?